test/addons/openssl-client-cert-engine/test.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+const common = require('../../common');
+const fixture = require('../../common/fixtures');
+
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const fs = require('fs');
+const path = require('path');
+
+const engine = path.join(__dirname,
+                         `/build/${common.buildType}/testengine.engine`);
+
+if (!fs.existsSync(engine))
+  common.skip('no client cert engine');
```

Comment:
Should this check be here?  Shouldn't that be an outright error?

Reply:
On CI, the necessary `libopenssl.a` file is not always there. So in that case, we don't build the engine. This is the test's way of detecting that situation. (In fact, the only host where this test is run on CI is macOS.)